docs/quickstarts/tanstack-start.mdx

```diff
@@ -0,0 +1,246 @@
+---
+title: Use Clerk with TanStack Start
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your TanStack Start application.
+---
+
+> [!WARNING]
+> The TanStack Start SDK is currently in beta. **It is not yet recommended for production use**.
+
+<TutorialHero
+  framework="react"
+  exampleRepo={[
+    {
+      title: "TanStack Start Quickstart Repo",
+      link: "https://github.com/clerk/clerk-tanstack-start-quickstart"
+
+    }
+  ]}
+  beforeYouStart={[
+    {
+      title: "Set up a Clerk application",
+      link: "/docs/quickstarts/setup-clerk",
+      icon: "clerk",
+    }
+  ]}
+>
+  - Install `@clerk/tanstack-start`
+  - Set up your environment variables
+  - Configure clerkHandler
+  - Add `<ClerkProvider/>` to your application
+  - Protecting your pages
+</TutorialHero>
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your TanStack Start application.
+
+<Steps>
+  ### Install `@clerk/tanstack-start`
+
+  Once you have a TanStack Start application ready, you need to install Clerk's TanStack Start SDK. This gives you access to our prebuilt components and hooks for TanStack Start applications.
+
+  <CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ prettier: false, filename: 'terminal' }}
+    npm install @clerk/tanstack-start
+    ```
+
+    ```bash {{ prettier: false, filename: 'terminal' }}
+    yarn add @clerk/tanstack-start
+    ```
+
+    ```bash {{ prettier: false, filename: 'terminal' }}
+    pnpm add @clerk/tanstack-start
+    ```
+  </CodeBlockTabs>
+
+  ### Set environment variables
+
+  Below is an example of an `.env` file.
+
+  **Pro tip!** If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can find your keys in the Clerk Dashboard on the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page.
+
+  ```env {{ prettier: false, filename: '.env' }}
+  VITE_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  ### Add `clerkHandler` to your application
+
+  To configure Clerk in your TanStack application, you will need to update SSR entrypoint.
+
+  ```tsx {{ prettier: false, filename: 'app/ssr.tsx' }}
```

Comment:
How does the default `app/ssr.tsx` file look like? Comparing it to that we can use line highlighting to outline the changes better